src/Util/Casacore.cc

```diff
@@ -257,6 +265,15 @@ void NormalizeUnit(casacore::String& unit) {
     }
 }
 
+bool IsGildasUnit(const casacore::String& unit) {
+    // Check if unit is in format from GILDAS CLASS software:
+    //     '<unit> (T<type>)' where the <type> describes the temperature T.
+    // Also test for casacore::Unit::check changes to the unit name where ' ' and '*' are replaced with '.'.
+    // For example: "K (Ta*)" -->  "K.(Ta.)" in casacore.
+    std::regex gildas_regex(GILDAS_UNIT_REGEX);
```

Comment:
Sorry, I meant defining the whole regex object outside, not just the string -- so that the object is created once, and not recreated whenever the function is called.